src/pynxtools_raman/reader.py

```diff
@@ -1,300 +1,243 @@
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""An example reader implementation based on the MultiFormatReader."""
+
+import copy
+import logging
+import datetime
+from typing import Dict, Any
+from pathlib import Path
+from typing import Any, Dict, List, Tuple  # Optional, Union, Set
+
+from pynxtools.dataconverter.readers.multi.reader import MultiFormatReader
+from pynxtools.dataconverter.readers.utils import parse_yml
+
+
+from pynxtools_raman.rod.rod_reader import RodParser
+from pynxtools_raman.witec.witec_reader import post_process_witec
+from pynxtools_raman.witec.witec_reader import parse_txt_file
+
+
+logger = logging.getLogger("pynxtools")
+
+CONVERT_DICT: Dict[str, str] = {}
+
+REPLACE_NESTED: Dict[str, str] = {}
+
+
+class RamanReader(MultiFormatReader):
+    """MyDataReader implementation for the DataConverter to convert mydata to NeXus."""
+
+    supported_nxdls = ["NXraman"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.raman_data_dicts: List[Dict[str, Any]] = []
+        self.raman_data: Dict[str, Any] = {}
+        self.eln_data: Dict[str, Any] = {}
+        self.config_file: Path
+
+        self.missing_meta_data = None
+
+        self.extensions = {
+            ".yml": self.handle_eln_file,
+            ".yaml": self.handle_eln_file,
+            ".txt": self.handle_txt_file,
+            ".json": self.set_config_file,
+            ".rod": self.handle_rod_file,
+        }
+
+        # only required if multiple file types are present
+        # for ext in RamanReader.__prmt_file_ext__:
+        #    self.extensions[ext] = self.handle_data_file
+
+    def set_config_file(self, file_path: Path) -> Dict[str, Any]:
+        if self.config_file is not None:
+            logger.info(
+                f"Config file already set. Replaced by the new file {file_path}."
+            )
+        self.config_file = file_path
+        return {}
+
+    def handle_eln_file(self, file_path: str) -> Dict[str, Any]:
+        self.eln_data = parse_yml(
+            file_path,
+            convert_dict=CONVERT_DICT,
+            parent_key="/ENTRY[entry]",
+        )
+
+        return {}
+
+    def get_attr(self, key: str, path: str) -> Any:
+        """
+        Get the metadata that was stored in the main file.
+        """
+        return self.get_metadata(self.raman_data, path, self.callbacks.entry_name)

```

Comment:
Are you using the `@attrs` prefix? If not, you should remove this function (especially since you don't have a `get_metadata` method defined).

Reply:
removed